src/app/services/actions/permissions-action-creator.ts

```diff
@@ -9,47 +13,60 @@ export function fetchScopesSuccess(response: object): IAction {
   };
 }
 
+export function fetchScopesPending(): any {
+  return {
+    type: FETCH_SCOPES_PENDING,
+  };
+}
+
 export function fetchScopesError(response: object): IAction {
   return {
     type: FETCH_SCOPES_ERROR,
     response,
   };
 }
 
-export function fetchScopes(): Function {
+export function fetchScopes(query?: IQuery): Function {
   return async (dispatch: Function, getState: Function) => {
-    const { sampleQuery: { sampleUrl, selectedVerb } } = getState();
-    const urlObject: URL = new URL(sampleUrl);
-    const createdAt = new Date().toISOString();
-    // remove the prefix i.e. beta or v1.0 and any possible extra '/' character at the end
-    const requestUrl = urlObject.pathname.substr(5).replace(/\/$/, '');
-    const permissionsUrl = 'https://graphexplorerapi.azurewebsites.net/api/GraphExplorerPermissions?requesturl=' +
-      requestUrl + '&method=' + selectedVerb;
-
-    const headers = {
-      'Content-Type': 'application/json',
-    };
-
-    const options: IRequestOptions = { headers };
-
-    return fetch(permissionsUrl, options)
-      .then(res => res.json())
-      .then(res => {
-        if (res.error) {
-          throw (res.error);
+    try {
+      const { devxApi } = getState();
```

Comment:
>devxApi [](start = 14, length = 7)

NIT: graphAPI as that's how it's being referred to going forward

Reply:
@ddyett interesting, graphAPI sounds like Microsoft Graph API can be confusing in the future.